### DISCUSSION TITLE
CouchDB performance issues in our biggest clusters

### DISCUSSION THREAD

**Discussion by ImreLbdv** [apache/couchdb#3591] [Q&A]
Hello,

We are actively using CouchDB in our projects and we've come across an issue that we can't get ourselves on top of. We're having three issues with CouchDB:

1) CouchDB runs out of memory
2) CouchDB CPU usage is very high at times
3) CouchDB increasingly taking a lot of space - _This is where our solution has been by deleting the data and letting it resync back from the other nodes. Which win us about 40G back every time. Downside to this, is that this causes some disturbance for the service. We do have smoosh in place and service does view cleanups from their own side as well, but that doesn't seem to have any impact on disk space_

Question is, if with following configuration, could we still have something to change from infrastructure side or some docs limiting needs to be done do solve our issues. Here's our setup:

```
CouchDB Version: 3.1
Nodes in cluster: 7
CPU per node: 8
RAM per node: 16
DB Count: 14100
Total Docs Count: 40834210
Total Docs size: 121,5 GB
_revs_limit: 1 per each db
```

**local.ini**

```
[cluster]
q = 8
n = 3

[couchdb]
max_document_size = 4294967296
max_attachment_chunk_size = 4294967296
os_process_timeout = 5000
max_dbs_open = 45000
open_os_files = 45000
delayed_commits = true
batch_save_size = 1000
batch_save_interval = 1000
single_max_doc_size = 4294967296

[chttpd]
port = 5984
bind_address = 0.0.0.0

[ahttpd]
port = 5986

[httpd]
max_http_request_size = 4294967296

[log]
level = error
writer = file
path = /var/log/couchdb/couchdb.log

[database_compaction]
doc_buffer_size = 10000000
checkpoint_after = 1048576

[smoosh]
db_channels = upgrade_dbs,ratio_dbs,slack_dbs
view_channels = upgrade_views,ratio_views,slack_views

[smoosh.ratio_dbs]
priority = ratio
min_priority = 1.25

[smoosh.ratio_views]
priority = ratio
min_priority = 1.25

[smoosh.slack_dbs]
priority = slack
min_priority = 200

[smoosh.slack_views]
priority = slack
min_priority = 200

[admins]
<admin-user> = <PW>

[config]
ahttpd = {"port"=>5984}

```

**vm.args:**
```
# CouchDB Configuration Settings

-name couchdb@<IP>

# All nodes must share the same magic cookie for distributed Erlang to work.
# Comment out this line if you synchronized the cookies by other means (using
# the ~/.erlang.cookie file, for example).
-setcookie monster

# Tell kernel and SASL not to log anything
-kernel error_logger silent
-sasl sasl_error_logger false

# Use kernel poll functionality if supported by emulator
+K true

# Start a pool of asynchronous IO threads
+A 16

# Comment this line out to enable the interactive Erlang shell on startup
+Bd -noinput
```

Thank you in advance for all the help and if I can provide anything else to help you understand the issues further, please do let me know.

**Comment by ImreLbdv** [comment]
We are still struggiling to find any solutions. Would be greatly appreciated if anyone has some ideas :) Thanks

**Comment by janl** [comment]
Suggested on Slack to look for conflicted docs and increase smoosh slack from 200 (bytes!) to the default: https://github.com/apache/couchdb/blob/3.x/rel/overlay/etc/default.ini#L633
